hoomd/update/box_resize.py

```diff
@@ -44,30 +45,34 @@ class BoxResize(Updater):
         variant (hoomd.variant.Variant): A variant used to interpolate between
             the two boxes.
         trigger (hoomd.trigger.Trigger): The trigger to activate this updater.
-        scale_particles (bool): Whether to scale particles to the new box
-            dimensions when the box is resized.
+        scale_particles (hoomd.filter.ParticleFilter): The subset of particles to
+            update.
     """
     def __init__(self, box1, box2,
-                 variant, trigger, scale_particles=True):
+                 variant, trigger, scale_particles=All()):
         params = ParameterDict(
             box1=OnlyType(Box, preprocess=box_preprocessing),
             box2=OnlyType(Box, preprocess=box_preprocessing),
             variant=Variant,
-            scale_particles=bool)
+            scale_particles=ParticleFilter)
         params['box1'] = box1
         params['box2'] = box2
         params['variant'] = variant
         params['trigger'] = trigger
         params['scale_particles'] = scale_particles
         self._param_dict.update(params)
+        self.scale_particles = scale_particles
```

Comment:
If the argument were just called `filter`, there is nothing in the name that signifies what it does:
`BoxResize(filter=some_filter)`. Does that mean that `BoxResize` is the wrong name for the class?

Reply:
Hmm, I am okay with this for now as `scale_particles` is more explicit even if somewhat inconsistent. Alternatively we could call it `SystemResize` to imply (correctly) that this can scale both the box and particles.

Reply:
I like the idea of changing the class name to `SystemResize` with a `filter` argument since it allows for consistency while remaining descriptive.

Reply:
Depends on how willing we are to break APIs that users are actively using with the betas.

Reply:
Or we can just change the argument to `filter` for consistency. The parameter will be properly documented, it isn't strictly necessary that the single line function call signature is self-explanatory.

Reply:
I switched the argument name from `scale_particles` to `filter`. Please let me know if there are other changes you would like me to make.